### DISCUSSION TITLE
Problem with DAW playing notes simultaneously on multiple midi channels. "Running Status" issue ?

### DISCUSSION THREAD

**Discussion by SurvivalHacking** [probonopd/MiniDexed#283] [Q&A] [answered]
I have a very strange problem that I can't figure out whether it depends on the "Running status" or something else. If I use the mini dexed by driving it via hardware midi from my keystation apparently I have no problem. Everything sounds with correct polyphony.
The same if I go through Logic, But if I record any polyphonic musical phrase, once it is quantized, it plays in a very peculiar way. Basically if there are several simultaneous notes, only one note is ever heard. This also develops in multi track. If in an 8-track, multitimbral project, a simultaneous NOTE-ON occurs on several tracks, only one track sounds. Basically, it seems that Logic somehow 'filters' the MIDI stream and the mini-dexed does not work properly.
If I put a track delay of 10ms on track 1 and 20ms on track 2 etc etc, it more or less plays everything, but obviously with noticeable delays. 
It is the simultaneity that is the real problem.

Does anyone have any suggestions on how to possibly set up Logic or mini-dexed to work properly ?

Can be a "Running Status" issue ?

Thanks

**Comment by Banana71** [comment]
Can you test it with USB Midi? 

For me it's sounds similar to this Problem
https://github.com/probonopd/MiniDexed/discussions/281

**Comment by SurvivalHacking** [comment]
I use a MIDI shield, and yes I use a cheap USB MIDI interface.
I need to test with a better USB midi interface I guess.

So I checked with oscilloscope the signal, and seem perfect.
I have not note loss, when use the m-audio  master keyboard. So after recording a track on Logic, and qunatize it, I ear only one note of multi note chord, exactly as the old analog keyboard where "running status" is not implemented.
Also sometime I have some note hang due loosing NOTE-OFF. 

Seems really a "Running Status" issue.



https://user-images.githubusercontent.com/76624331/174457398-ced1633b-4cfb-4a94-affd-d73b691125b0.mov

![Piastra-adattatore-interfaccia-digitale-strumento-scheda-Breakout-musicale-MIDI-Shield-per-modulo-scheda-adattatore-Arduino-compatibile](https://user-images.githubusercontent.com/76624331/174456512-b0e8bfcd-6830-46ea-b99e-1a2c803e7fe1.jpg)
![817LeQV3OKL _AC_SL1500_](https://user-images.githubusercontent.com/76624331/174456521-9b11bf46-505f-44c2-bff9-4633da7549f4.jpg)

**Comment by Banana71** [comment]
I stress tested my miniDexed and got good results.
I created eight midi tracks in Cubase. I chose my Arturia Keystep as the midi output. It only worked properly when I activated Midi Thru in the Keystep. Every note sounds on the right place. I think the problem is not with the miniDexed.

![Stresstest](https://user-images.githubusercontent.com/104296205/174469187-9771613b-c351-4d7a-a34f-44fd9a9c6a85.jpg)
.

**Reply by SurvivalHacking** [reply]
Here, this continues to increase my doubts about the correct implementation of "Running status" on the MiniDexed. Probably the Arturia's Thru is a 'Soft' Thru and a filter is applied that turns the MIDI stream into a stream without the handling of the "Running Status" and that is why everything works smoothly.

I am unfortunately not very skilled in programming and cannot understand if in the serialmididevice.cpp code the implementation is correct, but there does not seem to be any handling, although there is a comment mentioning it.

It would take some experts.

**Comment by Banana71** [comment]
I tested another old MIDI interface and connected it directly between my PC and the miniDexed.
Everything works as it should.

![OldUSBMidiInterface](https://user-images.githubusercontent.com/104296205/174470411-2400db6c-3438-45ab-ba8e-c1051c47cb39.jpg)

**Reply by SurvivalHacking** [reply]
Ok so I need to check if the problem can be my Midi interface or Apple Logic or other.

**Reply by SurvivalHacking** [reply]
Ok I tried with cubase and the result is the same. So The last thing I can try is another MIDI interface.
I bought one on AMZ... When come I post the result.
So I'm pretty sure that the problem was on bad "Running Status" management. The midi interface, is almost transparent and cannot change the behaviour on "Running Status" it reflect exactly what DAW send.

In a very OLD cubase version I remember that the "Running Status" can be disabled for compatibility issues on Old Analog Synth, is a good try if I find it.

**Reply by Banana71** [reply]
Have you checked the voltages on your MIDI shield? 5V, 3.3V are on the right PIN's?

**Comment by SurvivalHacking** [comment, marked as answer]
OK, passing through the M-Audio keyboard which I didn't know had MIDI from USB and could act as a MIDI card, it works great. So it is the cheap USB MIDI card that has some problem. 
Damn Usb midi !!!

https://user-images.githubusercontent.com/76624331/174479156-df87b728-0a8a-4c23-ad33-afb06fba8401.mov
